PetrolEngine.cpp

```diff
@@ -15,6 +13,15 @@ void PetrolEngine::changeGear(int gear)
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
+    if(currentGear_ > 0 && gear == -1)
+    {
+        InvalidGear exception("It is not possible to change gear from higher than 0 to R");
+        throw exception;
```

Comment:
```suggestion
        throw InvalidGear{"It is not possible to change gear from higher than 0 to R"}; 
```
It's a good practice to create unnamed temporary exceptions. You'll not be able to somehow modify them by name.